include/bluetooth/mesh/sensor_srv.h

```diff
@@ -58,6 +58,38 @@ struct bt_mesh_sensor_srv;
 					      _srv),                           \
 			 &_bt_mesh_sensor_setup_srv_cb)
 
+#ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV
+/** @def BT_MESH_MODEL_SENSOR_SRV_WITH_METADATA
+ *
+ *  @brief Sensor Server model composition data entry with metadata.
+ *
+ *  @param[in] _srv Pointer to a @ref bt_mesh_sensor_srv instance.
+ *  @param[in] _metadata Pointer to a @ref bt_mesh_models_metadata_entry.
+ */
+#define BT_MESH_MODEL_SENSOR_SRV_WITH_METADATA(_srv, _metadata)                                    \
+	BT_MESH_MODEL_METADATA_CB(BT_MESH_MODEL_ID_SENSOR_SRV, _bt_mesh_sensor_srv_op,             \
+				  &(_srv)->pub,                                                    \
+				  BT_MESH_MODEL_USER_DATA(struct bt_mesh_sensor_srv, _srv),        \
+				  &_bt_mesh_sensor_srv_cb, &_metadata),                     \
```

Comment:
Parenthesis everywhere and remove `&` so it gives more flexibility for a user:
```suggestion
				  &_bt_mesh_sensor_srv_cb, (_metadata)),                     \
```

Reply:
The design of access layer metadata handler implementation assumed the metadata pointer will be stored somewhere in the model. Like it has been done in Health server: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/bluetooth/mesh/health_srv.h#L160-L163
Since the same idea for NCS models was disagreed, then it requires providing pointer to pointer directly to access layer.
I do not think this is convenient for user to provide pointer to pointer directly in macros. It looks cumbersome.
Even now, it is difficult to explain why user needs to have both metadata and pointer to it. 
Nevertheless, the current implementation is more convenient if we rely on customer responsibility of managing both metadata and its pointer.

Reply:
The problem really in the double pointer. It should've been single pointer, then it would work like for opcode handlers. But now we need to have the extra pointer for unknown reasons.

I still lean towads removing `&` as the customer could do:
```
static struct bt_mesh_models_metadata_entry **entries = {
        &(struct bt_mesh_models_metadata_entry){...},
        &(struct bt_mesh_models_metadata_entry){...},
        ...
};
```

Reply:
I guess your example is against of Mesh specification. 
```
4.2.50.1 
Models Metadata Page 0 

Composition Metadata Page 0 shall not change during a term of a node on the network
```
Technically, your suggestion is possible but doesn't have sense.
I do not see any reason to have an implementation that breaks specification.

Reply:
Though I don't see contradiction in my suggestion with the spec, I will address only one point in your comment: you can change metadata between terms. For example, you can change the light purpose while device is unprovisioned.

Reply:
This is obvious. Why does the model require an array of metadata entries if specification mandates to use only one entry?
For changing of light purpose, you do not need to provide metadata that is not used. Just provide actual entry. That's it.

Reply:
I think the argue went away from the initial comment. I'm tired arguing here. I gave my reasons. Please remove `&`. Thanks.

Reply:
I provided my counterarguments against it. The reason of changes `because I want` doesn't seem correct to me. Thanks

Reply:
It is not `I want` it is about more flexibility. Please read my comment correctly.

Reply:
Yep, the same from me. I provided my counterarguments against of it. Those were ignored. Please read them.

Reply:
You get the same code as before, just pass `p_lightness_srv_metadata` with `&`. But then you also able to instantiate metadata entries differently (I gave example). I don't think we should limit the customers in how they should instantiate metadata.

Reply:
> Customer is able to instantiate metadata in any way (even how you provided in example). It will work in both implementations.

With your current changes I can't do this:

```
diff --git a/samples/bluetooth/mesh/light_ctrl/src/model_handler.c b/samples/bluetooth/mesh/light_ctrl/src/model_handler.c
index cbba6743bc2..508da22479c 100644
--- a/samples/bluetooth/mesh/light_ctrl/src/model_handler.c
+++ b/samples/bluetooth/mesh/light_ctrl/src/model_handler.c
@@ -107,6 +107,7 @@ static uint8_t health_tests[] = {
        BT_MESH_HEALTH_TEST_INFO(CONFIG_BT_COMPANY_ID, 3, 0x11, 0x22, 0x33),
 };
 
+#if 0
 static struct bt_mesh_models_metadata_entry lightness_srv_metadata[] = {
        BT_MESH_LIGHT_PURPOSE_METADATA(BT_MESH_LIGHT_PURPOSE_INDICATOR_LIGHT),
        BT_MESH_LIGHT_LIGHTNESS_RANGE_METADATA(0, UINT16_MAX),
@@ -114,6 +115,13 @@ static struct bt_mesh_models_metadata_entry lightness_srv_metadata[] = {
 };
 
 static struct bt_mesh_models_metadata_entry *p_lightness_srv_metadata = lightness_srv_metadata;
+#else
+static struct bt_mesh_models_metadata_entry *p_lightness_srv_metadata[] = {
+       &(struct bt_mesh_models_metadata_entry)BT_MESH_LIGHT_PURPOSE_METADATA(BT_MESH_LIGHT_PURPOSE_INDICATOR_LIGHT),
+       &(struct bt_mesh_models_metadata_entry)BT_MESH_LIGHT_LIGHTNESS_RANGE_METADATA(0, UINT16_MAX),
+       &(struct bt_mesh_models_metadata_entry)BT_MESH_MODELS_METADATA_END,
+};
+#endif
 
 static struct bt_mesh_models_metadata_entry health_srv_metadata[] = {
        BT_MESH_HEALTH_TEST_INFO_METADATA(health_tests),
```

```
[1/8] Building C object CMakeFiles/app.dir/src/model_handler.c.obj                                                                                                                                 [161/161]
In file included from /home/pavel/nordic/ncs/git-repos/ncs/zephyr/include/zephyr/bluetooth/bluetooth.h:24,                                                                                                  
                 from ../src/model_handler.c:7:                                                                                                                                                             
/home/pavel/nordic/ncs/git-repos/ncs/nrf/include/bluetooth/mesh/lightness_srv.h:82:53: warning: initialization of 'struct bt_mesh_models_metadata_entry **' from incompatible pointer type 'struct bt_mesh_m
odels_metadata_entry * (*)[3]' [-Wincompatible-pointer-types]                                                                                                                                               
   82 |                         &_bt_mesh_lightness_srv_cb, &_metadata),                 
```

> I try to convey idea to you about useless of providing pointer to pointer since the only single entry is used in access....

Just see how `BT_MESH_MODEL_OP_END` macro works:
https://github.com/zephyrproject-rtos/zephyr/blob/35176b993d4e06cc840b8dff61c905ba8437de50/include/zephyr/bluetooth/mesh/access.h#L394-L395
https://github.com/zephyrproject-rtos/zephyr/blob/35176b993d4e06cc840b8dff61c905ba8437de50/samples/bluetooth/mesh/src/main.c#L215-L219

Reply:
> With your current changes I can't do this:

With my suggestion I can initialize metadata in both ways.